src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java

```diff
@@ -99,49 +108,138 @@ private GenerationModel mockGenerationModel(String example) {
+      assertEquals("httpListenerConfig", consoleFlow.getChildren().get(0).getAttribute("config-ref").getValue());
+      assertEquals("/console/*", consoleFlow.getChildren().get(0).getAttribute("path").getValue());
+      assertEquals("console", consoleFlow.getChildren().get(1).getName());
+    }
   }
 
   @Test
   public void blankDocumentWithoutLCInDomainHideConsole() {
-    Document document = scaffoldBlankDocument(HIDE_CONSOLE);
-    Element rootElement = document.getRootElement();
-    assertCommonFlows(rootElement);
-
+    ScaffoldingConfiguration.Builder builder = ScaffoldingConfiguration.builder();
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      assertCommonFlows(rootElement);
+    }
   }
 
   private void assertCommonFlows(Element rootElement) {
     assertEquals("mule", rootElement.getName());
-    Element xmlListenerConfig = rootElement.getChildren().get(0);
-    assertEquals("listener-config", xmlListenerConfig.getName());
+    assertConfigurations(rootElement);
 
-    Element apikitConfig = rootElement.getChildren().get(1);
-    assertEquals("hello-config", apikitConfig.getAttribute("name").getValue());
+    assertMainFlow(rootElement, 2);
+  }
 
-    Element mainFlow = rootElement.getChildren().get(2);
+  private void assertMainFlow(Element rootElement, int index) {
+    Element mainFlow = rootElement.getChildren().get(index);
 
     assertEquals("flow", mainFlow.getName());
     assertEquals("hello-main", mainFlow.getAttribute("name").getValue());
     assertEquals("httpListenerConfig", mainFlow.getChildren().get(0).getAttribute("config-ref").getValue());
     assertEquals("/api/*", mainFlow.getChildren().get(0).getAttribute("path").getValue());
   }
 
-  private Document scaffoldBlankDocument(boolean showConsole) {
+  private void assertConfigurations(Element rootElement) {
+    Element xmlListenerConfig = rootElement.getChildren().get(0);
+    assertEquals("listener-config", xmlListenerConfig.getName());
+
+    Element apikitConfig = rootElement.getChildren().get(1);
+    assertEquals("hello-config", apikitConfig.getAttribute("name").getValue());
+  }
+
+  private List<MuleConfig> scaffoldBlankDocument(ScaffoldingConfiguration configuration) {
     HttpListenerConfig listenerConfig =
         new HttpListenerConfig(HttpListenerConfig.DEFAULT_CONFIG_NAME, "localhost", "8080", "HTTP", "");
+    InputStream apikitConfig = getResourceAsStream("scaffolder-only-apikit-config/api.xml");
+    MuleConfig muleConfig = buildAPIKitMuleConfig(apikitConfig);
     ApikitMainFlowContainer api = mock(ApikitMainFlowContainer.class);
     when(api.getPath()).thenReturn("/api/*");
     when(api.getHttpListenerConfig()).thenReturn(listenerConfig);
+    when(mock(MuleConfig.class).getApikitConfigs()).thenReturn(muleConfig.getApikitConfigs());
```

Comment:
This is never going to be called. You are mocking the response of a mock. I think what you wanted to do is 
```when(api.getMuleConfig()).thenReturn(muleConfig);```
Right?